pkg/resources/istiod/deployment.go

```diff
@@ -290,7 +357,39 @@ func (r *Reconciler) volumes() []apiv1.Volume {
 			})
 		}
 
-		if r.Config.Spec.Citadel.CASecretName != "" {
+		if util.PointerToBool(r.Config.Spec.Istiod.CA.Vault.Enabled) {
+			volumes = append(volumes, []apiv1.Volume{
+				{
+					Name: "cacerts",
+					VolumeSource: apiv1.VolumeSource{
+						EmptyDir: &apiv1.EmptyDirVolumeSource{},
+					},
+				},
+				{
+					Name: "vault-tls",
```

Comment:
Nit: String variables could be used for values, which are used multiple places e.g. volume names, env var names.

Reply:
👍